asv_bench/benchmarks/omnisci/benchmarks.py

```diff
@@ -0,0 +1,163 @@
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""General Modin on OmniSci backend benchmarks."""
+
+
+from ..utils import (
+    generate_dataframe,
+    RAND_LOW,
+    RAND_HIGH,
+    ASV_USE_IMPL,
+    ASV_DATASET_SIZE,
+    execute,
+)
+
+BINARY_OP_DATA_SIZE = {
+    "big": [
+        ((500_000, 20), (1_000_000, 10)),
+    ],
+    "small": [
+        ((10_000, 20), (25_000, 10)),
+    ],
+}
+
+UNARY_OP_DATA_SIZE = {
+    "big": [
+        (1_000_000, 10),
+    ],
+    "small": [
+        (10_000, 10),
+    ],
+}
+
+
+def trigger_import(dfs: list):
+    from modin.experimental.engines.omnisci_on_ray.frame.omnisci_worker import (
+        OmnisciServer,
+    )
+
+    for df in dfs:
+        df.shape  # to trigger real execution
+        df._query_compiler._modin_frame._partitions[0][
+            0
+        ].frame_id = OmnisciServer().put_arrow_to_omnisci(
+            df._query_compiler._modin_frame._partitions[0][0].get()
+        )  # to trigger real execution
+
+
+class TimeMerge:
+    param_names = ["shapes", "how"]
+    params = [
+        BINARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        ["left"],
+    ]
+
+    def setup(self, shapes, how):
+        self.df1 = generate_dataframe(
+            ASV_USE_IMPL, "int", *shapes[0], RAND_LOW, RAND_HIGH
+        )
+        self.df2 = generate_dataframe(
+            ASV_USE_IMPL, "int", *shapes[1], RAND_LOW, RAND_HIGH
+        )
+        trigger_import((self.df1, self.df2))
+
+    def time_merge(self, shapes, how):
+        # merge dataframes on index to get the predictable shape doesn't support
+        execute(
+            self.df1.merge(self.df2, on="col1", how=how, suffixes=("left_", "right_"))
+        )
+
+
+class TimeIndexing:
+    param_names = ["shape", "indexer_type"]
+    params = [
+        UNARY_OP_DATA_SIZE[ASV_DATASET_SIZE],
+        [
+            "scalar",
+            "bool",
+            "slice",
+            "list",
+            "function",
+        ],
+    ]
+
+    def setup(self, shape, indexer_type):
+        self.df = generate_dataframe(ASV_USE_IMPL, "int", *shape, RAND_LOW, RAND_HIGH)
+        trigger_import((self.df,))
+        if indexer_type == "bool":
+            self.indexer = [False, True] * (shape[0] // 2)
+        elif indexer_type == "scalar":
+            self.indexer = shape[0] // 2
+        elif indexer_type == "slice":
+            self.indexer = slice(0, shape[0], 2)
+        elif indexer_type == "list":
+            self.indexer = [x for x in range(shape[0])]
+        elif indexer_type == "function":
+            self.indexer = lambda df: df.index[::-2]
```

Comment:
```suggestion
        self.indexer = {
            "bool": [False, True] * (shape[0] // 2),
            "scalar": shape[0] // 2,
            "slice": slice(0, shape[0], 2),
            "list": list(range(shape[0])),
            "function": lambda df: df.index[::-2]
        }[indexer_type]
```
this is shorter and (IMHO) easier to read version with a built-in check for unknown value of `indexer_type`